lib/src/main/java/tech/relaycorp/awaladroid/messaging/IncomingMessage.kt

```diff
@@ -37,23 +46,28 @@ public class IncomingMessage internal constructor(
             InvalidMessageException::class,
             SetupPendingException::class,
         )
-        internal suspend fun build(parcel: Parcel, ack: suspend () -> Unit): IncomingMessage {
+        internal suspend fun build(parcel: Parcel, ack: suspend () -> Unit): IncomingMessage? {
             val recipientEndpoint = FirstPartyEndpoint.load(parcel.recipientAddress)
                 ?: throw UnknownFirstPartyEndpointException(
-                    "Unknown third party endpoint with address ${parcel.recipientAddress}"
+                    "Unknown first-party endpoint ${parcel.recipientAddress}"
                 )
 
             val sender = ThirdPartyEndpoint.load(
                 parcel.recipientAddress,
                 parcel.senderCertificate.subjectPrivateAddress,
             ) ?: throw UnknownThirdPartyEndpointException(
-                "Unknown third party endpoint with address " +
+                "Unknown third-party endpoint " +
                     "${parcel.senderCertificate.subjectPrivateAddress} " +
-                    "for first party endpoint ${parcel.recipientAddress}"
+                    "for first-party endpoint ${parcel.recipientAddress}"
             )
 
             val context = Awala.getContextOrThrow()
             val serviceMessage = context.endpointManager.unwrapMessagePayload(parcel)
+            if (serviceMessage.type == PDA_PATH_TYPE) {
+                processPDAPath(serviceMessage.content, sender, recipientEndpoint)
+                ack()
+                return null
+            }
```

Comment:
This works here. I was wondering if ReceiveMessages wouldn't be a better place to handle this type of messages. Until now this `IncomingMessage` didn't have any side-effect, now it feels a bit weird that calling `IncomingMessage.build` actually saves certain messages.

Reply:
I considered that, but the "problem" is that `IncomingMessage.build()` takes a `Parcel` instance and then retrieves the sender/recipient endpoints from it... which we'd also need in this new function to process incoming PDAs.

So I think to do that we'd need to move this to `ReceiveMessages`:

https://github.com/relaycorp/awala-endpoint-android/blob/b19cffb3501cab507cbd036ef4b4bc8590e45dde/lib/src/main/java/tech/relaycorp/awaladroid/messaging/IncomingMessage.kt#L50-L65

Which, in turn, means we'd no longer need a static function to initalise an `IncomingMessage` so it could be a regular constructor call that takes the `serviceMessage`, `sender` and `receiver`.

I like that design better than the current implementation, but chose not to do it for expediency... Though I'm not sure it was a wise idea, so I'll change it.

Reply:
We can leave that as a future improvements as more message types need to be handled. I would suggest then a method name change "build" to "buildOrProcess".

Reply:
Yup, I was just about to suggest deferring it because it was looking like a big change at this point. Plus as you say it's likely that there will be more special service messages that shouldn't be propagated to the app in the future.